solr_helpers/tests.py

```diff
@@ -0,0 +1,1606 @@
+        for bucket in bucket_list:
+            if bucket['val'] == test_vals['MOD_TEST_VAL']:
+                found_bucket = True
+                self.assertEquals(bucket['count'], test_vals['MOD_TEST_COUNT'])
+                break
+        self.assertTrue(found_bucket)
+
+        self.assertTrue('Species' in facet_dict)
+        self.assertTrue('missing' in facet_dict['Species'])
+        missing = facet_dict['Species']['missing']
+        self.assertTrue('count' in missing)
+        self.assertEquals(missing['count'], test_vals['MISSING_SPEC'])
+        self.assertTrue('buckets' in facet_dict['Species'])
+        bucket_list = facet_dict['Species']['buckets']
+        self.assertEquals(len(bucket_list), test_vals['SPEC_BUCKET_COUNT'])
+        found_bucket = False
+        for bucket in bucket_list:
+            if bucket['val'] == test_vals['SPEC_TEST_VAL']:
+                found_bucket = True
+                self.assertEquals(bucket['count'], test_vals['SPEC_TEST_COUNT'])
+                break
+        self.assertTrue(found_bucket)
+
+        self.assertTrue('response' in query_result)
+        self.assertEquals(query_result['response']['numFound'], test_vals['TOTAL_SIZE'])
+        return
+
+    #
+    # Build a payload that asks for the full collection, plus archived response:
+    #
+
+    @staticmethod
+    def _build_full_collection_payload_and_response():
+
+        mod_dict = {"field": "Modality",
+                    "missing": True,
+                    "type": "terms",
+                    "limit": -1}
+
+        species_dict = {"field": "Species",
+                        "missing": True,
+                        "type": "terms",
+                        "limit": -1}
+
+        coll_dict = {"field": "collection_id",
+                     "missing": True,
+                     "type": "terms",
+                     "limit": -1}
+
+        bpex_dict = {"field": "BodyPartExamined",
+                     "missing": True,
+                     "type": "terms",
+                     "limit": -1}
+
+        payload = {
+            "query": "*:*",
+            "limit": 0,
+            "offset": 0,
+            "params": {"debugQuery": "off"},
+            "facet": {"Modality": mod_dict,
+                      "Species": species_dict,
+                      "collection_id": coll_dict,
+                      "BodyPartExamined": bpex_dict},
+            "filter": ["{!collapse field=PatientID}"],
+            "fields": ["collection_id", "case_barcode", "PatientID", "race", "vital_status",
+                             "ethnicity", "bmi", "age_at_diagnosis", "gender", "disease_code",
+                             "StudyInstanceUID", "StudyDescription", "StudyDate",
+                             "SeriesInstanceUID", "SeriesDescription", "SeriesNumber",
+                             "BodyPartExamined", "Modality"]
+        }
+
+        #
+        # Return value of dev SOLR server recorded on 2/29/19
+        #
+
```

Comment:
Rather than putting archived responses in the code, maybe we should just have some JSON files for that. This makes them a lot easier to update in the long run as we swap datasets. Thoughts?

Reply:
You have a point. I was trying to keep the file count and complexity to a minimum. If we do not change the functionality of the module, then it won't really matter now the datasets change, the unit tests don't necessarily need to keep up.

Reply:
I don't know that we CAN guarantee a lack of module change, is the thing. So having the data be a JSON file we load makes more sense than bloating the test with it.

Reply:
OK, I can move those out to JSON files if you prefer. Put them in a subdirectory?

Reply:
Yeah, probably best to do that; note you'll need to put exceptions for them into .gitignore.

Reply:
Not sure why we would be trying to keep them out of git? They gat recorded one, and then used again and again when the test runs, so we need to put them in the repo.

Reply:
The exception lets them be added, because we default deny JSON files (as the keyfiles tend to be JSON files). So !myfile.json will let it bypass the *.json denial.

Reply:
Ah, yes, right. All .json files excluded currently.